src/app/lockers/util.ts

```diff
@@ -0,0 +1,54 @@
+import type { LockerWithReservation } from '@/server/lockers/Types'
+
+export function getGroupNameFromLocker(locker: LockerWithReservation): string {
+    if (locker.LockerReservation.length === 0) {
+        return ''
+    }
+
+    const reservation = locker.LockerReservation[0]
+    if (!reservation.group) {
+        return ''
+    }
+
+    const group = reservation.group
+
+    return getGroupName(group)
+}
+
+// This should probably be in another file, but I don't know where
+export function getGroupName(group) {
```

Comment:
Its called inferGroupName or something in feat/visibility. It is in the server folder so it should not be called client side the idea is to add this name field to the model before it reaches the client (GroupExtended or something)